plutus-core/plutus-core/src/PlutusCore/Value.hs

```diff
@@ -470,3 +476,33 @@ updateSizes old new = dec . inc
       then id
       else IntMap.update (\n -> if n <= 1 then Nothing else Just (n - 1)) old
 {-# INLINEABLE updateSizes #-}
+
+-- | Scale each tokens by the given constant factor.
+scaleValue :: Integer -> Value -> BuiltinResult Value
+scaleValue c (Value outer sizes size neg)
+  -- When scaling by positive factor, no need to change sizes and number of negative amounts.
+  | c > 0 = do
+    outer' <- go outer
+    BuiltinSuccess $ Value outer' sizes size neg
+  -- When scaling by negative factor, only need to "flip" negative amounts.
+  | c < 0 = do
+    outer' <- go outer
+    -- let
+    --   neg' = Map.foldl' alg 0 outer'
+    --   alg n inner = n + Map.size (Map.filter (< zeroQuantity) inner)
+
+    -- TODO: make sure (size - neg) is correct
+    BuiltinSuccess $ Value outer' sizes size (size - neg)
```

Comment:
please check here!

Need to make sure if I can just `size - neg` here or if I need to count all negative quantities again.

Reply:
Okay, added bookkeeping property as well. It seems to be fine.